app/utility/base_planning_svc.py

```diff
@@ -113,6 +118,41 @@ async def obfuscate_commands(self, agent, obfuscator, links):
 
     """ PRIVATE """
 
+    @staticmethod
+    def _filter_links_by_trait(link, trait):
+        """
+        Filter Link based on presence of trait in facts used
+        """
+        if any(trait in y.trait for y in link.used):
+            return link
+
+    @staticmethod
+    def _list_historic_dup_lat_mov(operation):
+        lat = [BasePlanningService._filter_links_by_trait(k, 'remote.host.fqdn') for k in operation.chain
+               if k.status != k.states['ERROR']]
+        return [(x.command_hash if x.command_hash else x.command) for x in lat if x]
+
+    @staticmethod
+    def _cross_check_agents_for_duplicate_lat_mov(agent_links):
```

Comment:
always better to spell out than use abbreviations

Reply:
I don't agree. Function names shouldn't be full descriptors of what the function does, that's the purpose of the comment. A 51 character function name doesn't serve any useful purpose than cluttering up other parts of the code.